tests/cupyx_tests/scipy_tests/spatial_tests/test_distance.py

```diff
@@ -122,3 +122,75 @@ def test_minkowski_(self, xp, scp):
         a = self._make_matrix(xp, self.dtype, self.order)
         out = scp.spatial.distance.minkowski(a, a, p=self.p)
         return out
+
+    @testing.numpy_cupy_equal(scipy_name='scp')
+    def test_canberra_(self, xp, scp):
+
+        a = self._make_matrix(xp, self.dtype, self.order)
+        out = scp.spatial.distance.canberra(a, a)
+        return out
+
+    @testing.numpy_cupy_equal(scipy_name='scp')
+    def test_chebyshev_(self, xp, scp):
+
+        a = self._make_matrix(xp, self.dtype, self.order)
+        out = scp.spatial.distance.chebyshev(a, a)
+        return out
+
+    @testing.numpy_cupy_equal(scipy_name='scp')
+    def test_cityblock_(self, xp, scp):
+
+        a = self._make_matrix(xp, self.dtype, self.order)
+        out = scp.spatial.distance.cityblock(a, a)
+        return out
+
+    @testing.numpy_cupy_array_almost_equal(scipy_name='scp', type_check=False)
+    def test_correlation_(self, xp, scp):
+
+        a = self._make_matrix(xp, self.dtype, self.order)
+        out = scp.spatial.distance.correlation(a, a)
```

Comment:
What did you get for `out`? (I mean why we need `xp.asarray` in the next line and `type_check=False` in the decorator?)

Reply:
For `cosine`, `russellrao` and `correlation`, the scipy output type is `float32` (not `float64`) and the raft result is near zero, but not zero

Reply:
> raft result is near zero, but not zero

We can do a threshold for this on the RAFT side if it helps.

Reply:
Okay, I got it. Thanks!